include/godot_cpp/core/method_ptrcall.hpp

```diff
@@ -170,7 +170,7 @@ template <class T>
 struct PtrToArg<T *> {
 	static_assert(std::is_base_of<Object, T>::value, "Cannot encode non-Object value as an Object");
 	_FORCE_INLINE_ static T *convert(const void *p_ptr) {
-		return reinterpret_cast<T *>(godot::internal::get_object_instance_binding(*reinterpret_cast<GDExtensionObjectPtr *>(const_cast<void *>(p_ptr))));
+		return p_ptr ? reinterpret_cast<T *>(godot::internal::get_object_instance_binding(*reinterpret_cast<GDExtensionObjectPtr *>(const_cast<void *>(p_ptr)))) : nullptr;
```

Comment:
seems good, but I would use unlikely(p_ptr) since this is a very corner case and we don´t want the conditional to have a performance impact.

Reply:
Thanks!

We did discuss using `likely()` here on RocketChat, however, my worry at the time was that we don't really know if passing null for an `Object *` is that unlikely: there's lots of functions that have default arguments which are null. However, thinking about it some more, I think we can probably say that it's unlikely enough (at least when Godot is calling into a GDExtension), because otherwise people would be encountering this bug way more often than they do.

So, I'll update the PR adding `likely()` here!

But, if we're going to do that here, we should probably also do it in the `convert()` function a couple lines below my change, which also has this same check (used for when calling from GDExtension into Godot). _And_ we also have this same check in `PtrToArg` in Godot itself, which also isn't using `likely()` that we should probably update for consistency - I'll make a new PR to address that as well.